qiskit/optimization/problems/quadratic_program.py

```diff
@@ -152,36 +153,139 @@ def _add_variable(self,
                       upperbound: Union[float, int] = INFINITY,
                       vartype: VarType = VarType.CONTINUOUS,
                       name: Optional[str] = None) -> Variable:
-        """Checks whether a variable name is already taken and adds the variable to list and index
-        if not.
+        return self._add_variables(lowerbound, upperbound, vartype, name)[1][0]
+
+    def _add_variables(self,
+                       lowerbound: Union[float, int] = 0,
+                       upperbound: Union[float, int] = INFINITY,
+                       vartype: VarType = VarType.CONTINUOUS,
+                       name: Optional[str] = None,
+                       key_format: Optional[str] = None,
+                       keys: Union[int, Sequence] = 1) -> Tuple[List[str], List[Variable]]:
+        if isinstance(keys, int) and keys < 1:
+            raise QiskitOptimizationError(
+                "Cannot create non-positive number of variables: {}".format(keys))
+        if not name and key_format:
+            name = 'x'
+        if name and not key_format:
+            if isinstance(keys, int) and keys > 1 or isinstance(keys, Sequence) and len(keys) > 0:
+                key_format = '{}'
+            else:
+                key_format = ''
+                if name in self._variables_index:
+                    raise QiskitOptimizationError("Variable name already exists: {}".format(name))
+        if not name and not key_format:
+            name = 'x'
+            key_format = '{}'
+        if '{{}}' in key_format:
+            raise QiskitOptimizationError(
+                "Formatter cannot contain nested substitutions: {}".format(key_format))
+        substitution_count = key_format.count('{}')
```

Comment:
This check only needs to happen if a `key_format` has been provided. So as to not duplicate existence checks for `key_format`, it can go in the `not name and key_format` case.